query_result_test.go

```
@@ -0,0 +1,174 @@
+	_, err = QueryResultNew(graph, []interface{}{"header", "records"})
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "malformed response")
+}
+
+// End-to-end proof that a hostile map key cannot escape its literal. The
+// payload below used to create a `:Pwned` node and comment out the query.
+func TestQueryParamCannotInjectCypher(t *testing.T) {
+	db, err := FromURL("falkor://0.0.0.0:6379")
+	require.NoError(t, err)
+	victim := db.SelectGraph("injection_regression")
+	defer victim.Delete()
```

golangci-lint (2.12.2)

[error] Error return value of `victim.Delete` is not checked

(errcheck)